lib/dfe/analytics/requests.rb

```diff
@@ -16,7 +16,7 @@ def trigger_request_event
                                              .with_type('web_request')
                                              .with_request_details(request)
                                              .with_response_details(response)
-                                             .with_user_and_namespace(current_user, try(:current_namespace))
+                                             .with_user_and_namespace(try(:current_user), try(:current_namespace))
```

Comment:
probably nicer to go symmetry...

```suggestion
                                             .with_user(try(:current_user))
                                             .with_namespace(try(:current_namespace))
```

Reply:
I agree

Reply:
Ok, I'll split out these into two methods.

Reply:
I've done this, but also changed the code to be slightly more descriptive:

```ruby
        request_event.with_user(current_user)           if respond_to? :current_user
        request_event.with_namespace(current_namespace) if respond_to? :current_namespace
```

One effect this has is that if these methods aren't implemented then `user_id` and `namespace` won't be part of the payload to BQ. I don't think this will change any behaviour in BQ.